tests/Unit/Time/Actions/Test_LimitTimeStepToExpirationTimes.cpp

```diff
@@ -0,0 +1,175 @@
+           start.value(), initial_quaternion, initial_rotation,
+           before_expiration_times[1])});
+
+  using component = Component<Metavariables>;
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<Metavariables>;
+  MockRuntimeSystem runner{{std::make_unique<TimeStepper>()},
+                           {std::move(functions_of_time)}};
+  const TimeStepId& time_step_id{time_step.is_positive(), 8, start};
+  const TimeStepId& next_time_step_id{time_step.is_positive(), 9,
+                                      start + time_step};
+  ActionTesting::emplace_component_and_initialize<component>(
+      &runner, 0,
+      {time_step_id, time_step, start.value(), time_step, next_time_step_id});
+  ActionTesting::set_phase(make_not_null(&runner), Parallel::Phase::Testing);
+
+  runner.next_action<component>(0);
+
+  const auto& box = ActionTesting::get_databox<component>(runner, 0);
+  const auto& new_time_step = db::get<Tags::TimeStep>(box);
+  const auto& new_next_time_step = db::get<Tags::Next<Tags::TimeStep>>(box);
+  const auto& new_time_step_id = db::get<Tags::TimeStepId>(box);
+  const auto& new_next_time_step_id =
+      db::get<Tags::Next<Tags::TimeStepId>>(box);
+  const double new_slab_start_value{new_time_step.slab().start().value()};
+  const double new_slab_end_value{new_time_step.slab().end().value()};
+
+  CHECK(new_slab_start_value == time_step.slab().start().value());
+  CHECK((min_expiration_time == new_slab_end_value));
+
+  CHECK(new_slab_start_value + time_step.value() > min_expiration_time);
+  CHECK(new_slab_start_value + new_time_step.value() == min_expiration_time);
+
+  CHECK(new_time_step.fraction() == Rational{1, 1});
+  CHECK(new_time_step.slab().start().value() == new_slab_start_value);
+  CHECK(new_time_step.slab().end().value() == new_slab_end_value);
+
+  CHECK(new_time_step_id.time_runs_forward() ==
+        time_step_id.time_runs_forward());
+  CHECK(new_time_step_id.slab_number() == time_step_id.slab_number());
+  CHECK(new_time_step_id.step_time().fraction() ==
+        time_step_id.step_time().fraction());
+  CHECK(new_time_step_id.substep() == time_step_id.substep());
+  CHECK(new_time_step_id.substep_time().fraction() ==
+        time_step_id.substep_time().fraction());
+  CHECK(new_time_step_id.step_time().slab().start().value() ==
+        new_slab_start_value);
+  CHECK(new_time_step_id.step_time().slab().end().value() ==
+        new_slab_end_value);
+  CHECK(new_time_step_id.substep_time().slab().start().value() ==
+        new_slab_start_value);
+  CHECK(new_time_step_id.substep_time().slab().end().value() ==
+        new_slab_end_value);
+
+  CHECK(new_next_time_step_id == db::get<Tags::TimeStepper<>>(box).next_time_id(
+                                     new_time_step_id, new_time_step));
+  CHECK(new_next_time_step.fraction() == time_step.fraction());
+  CHECK(new_next_time_step.slab() == new_next_time_step_id.step_time().slab());
+}
```

Comment:
I'm wondering if you should add some tests for all the different `if` statements in your action. Codecov wasn't run, but I think it'd say some parts of the code aren't tested. Things like

- Timestepper doesn't have substeps
- We aren't at the beginning of a timestep
- We are more than 2 steps away from the expiration time
- We are less than 2 steps more than 1 step away
- We are less than one step away